### PR TITLE
Getty images plugin

### PR DESCRIPTION
**Pull request by elv1n**
Getty Images plugin and documentation for PhotoEditorSDK.

The live demo is going to be added on codesanbox after reviewing this PR and npm release.
*live demo still requires adding valid token*

### CODE REVIEW

**Comment by alanuecker** [comment]
@elv1n so I build the plugin in `getty-images/src` and followed the instructions in `example` but ran into the following issues:
image

File: plugins/getty-images/LICENCE.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+This sample code or wrapper is licensed with a 3-clause BSD license.
```
Comment (sascha) [inline plugins/getty-images/LICENCE.md]: Can you name the file `LICENSE.md`? That's how we've spelled it everywhere else I guess.

File: plugins/getty-images/README.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
+# Getty Images plugin for PhotoEditor SDK
+
+### Prerequisite
+
+This plugin helps to integrate Getty Images API into PhotoEditorSDK.
+
+You need to use own backend endpoint that provides [OAuth token](https://developer.gettyimages.com/docs/oauth-2.0/) for Getty Images API.
```
Comment (sascha) [inline plugins/getty-images/README.md]:
```suggestion
You need to use your own backend endpoint that provides an [OAuth token](https://developer.gettyimages.com/docs/oauth-2.0/) for the Getty Images API.
```

File: plugins/getty-images/README.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            {...props}
+            {...params}
+        />
+    );
+};
+```
+
+#### Customization
+
+You can replace toolbar using any library that you want or have in
```
Comment (sascha) [inline plugins/getty-images/README.md]:
```suggestion
You can replace the toolbar using any library that you want or have in
```

File: plugins/getty-images/README.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+the app environment.
+
+To override toolbar you must keep:
```
Comment (sascha) [inline plugins/getty-images/README.md]:
```suggestion
To override the toolbar you must keep:
```

File: plugins/getty-images/README.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+* createAPIClient instance
```
Comment (sascha) [inline plugins/getty-images/README.md]:
```suggestion
* `createAPIClient` instance
```

File: plugins/getty-images/README.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+* gettyStore calls
```
Comment (sascha) [inline plugins/getty-images/README.md]:
```suggestion
* `gettyStore` calls
```

File: plugins/getty-images/README.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+* `handleSetImage` function
+
+### Export button
+
+The export button should be used to get higher resolution and license Getty image.
```
Comment (sascha) [inline plugins/getty-images/README.md]:
```suggestion
The export button should be used to get a higher resolution and licensed image from Getty Images.
```

File: plugins/getty-images/README.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+        />
+    );
+};
+```
+
+#### Customization
+
+You can replace the export button with a custom one.
+
+To override toolbar you must keep:
```
Comment (sascha) [inline plugins/getty-images/README.md]:
```suggestion
To override the toolbar you must keep:
```

Is 'toolbar' correct here? Or did you mean to write 'To override the export button you must keep'?

File: plugins/getty-images/README.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+* Passing custom button props if `ContainedPrimaryButton` going to be used
```
Comment (sascha) [inline plugins/getty-images/README.md]:
```suggestion
* Passing custom button props if `ContainedPrimaryButton` is going to be used
```

File: plugins/getty-images/README.md [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+* `onLicenseImage` function
+
+
+## PhotoeditorSDK Documentation
+Visit our [docs](https://docs.photoeditorsdk.com)
```
Comment (sascha) [inline plugins/getty-images/README.md]:
```suggestion
## PhotoEditor SDK Documentation
Visit our [documentation](https://docs.photoeditorsdk.com).
```

File: plugins/getty-images/README.md [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+
+## Examples
+[PhotoeditorSDK & GettyImages example](https://github.com/imgly/pesdk-web-plugins/tree/main/plugins/getty-images/example)
+[PhotoeditorSDK examples](https://www.photoeditorsdk.com/)
```
Comment (sascha) [inline plugins/getty-images/README.md]:
```suggestion
[PhotoEditor SDK & Getty Images example](https://github.com/imgly/pesdk-web-plugins/tree/main/plugins/getty-images/example)
[PhotoEditor SDK examples](https://www.photoeditorsdk.com/)
```

**Review by sascha — CHANGES_REQUESTED** [review]

File: plugins/getty-images/src/GettyImagesToolControlBar.tsx [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+  SearchDiv,
+  Spinner,
+  ToolbarWrapper,
+} from './components/Elements';
+
+interface Props {
+  client: ReturnType<typeof createAPIClient>;
+  language: CustomToolProps['language'];
+  onError: OnError;
+  searchParams?: SearchImagesParams;
```
Comment (alanuecker) [inline plugins/getty-images/src/GettyImagesToolControlBar.tsx]: we should use `Partial` here, else `SearchImagesParams.phrase` is always required and would override the search field phrase

File: plugins/getty-images/src/GettyImagesToolControlBar.tsx [hunk 13]
```diff
@@ -0,0 +1,208 @@
...
+   * https://github.com/imgly/pesdk-web-plugins/tree/main/plugins/getty-images/src/types.ts
+   */
+  onError: OnError;
+  /**
+   * configure image search params
+   * Typescript types https://github.com/imgly/pesdk-web-plugins/tree/main/plugins/getty-images/src/api/searchImages.ts
+   * Getty API https://api.gettyimages.com/swagger/index.html#Images
+   */
+  searchParams?: SearchImagesParams;
+  // image size for editor preview, default DisplaySizeName.High = 'high_res_comp'
```
Comment (alanuecker) [inline plugins/getty-images/src/GettyImagesToolControlBar.tsx]: Seems like normal comments `//` will be stripped from the `.d.ts` files.

File: plugins/getty-images/src/helpers.ts [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+  image: GettyImage,
+  sizeName: DisplaySizeName,
+): DisplaySize | undefined =>
+  image.display_sizes.find(i => i.name === sizeName);
+
+// calculate received search results and check if more items available to request next page
+export const hasNextPage = (
+  result: SearchImagesResponse[],
+  maxPerPage: number,
+) => {
```
Comment (alanuecker) [inline plugins/getty-images/src/helpers.ts]: eslint complained about the missing return type

File: plugins/getty-images/src/helpers.ts [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+type GettyStore = {
+  img?: GettyImage;
+  client?: ReturnType<typeof createAPIClient>;
+};
+
+let storeData: GettyStore = {};
+
+export const gettyStore = {
```
Comment (alanuecker) [inline plugins/getty-images/src/helpers.ts:57]: eslint complained about missing return type for both functions

File: plugins/getty-images/src/hooks/useIntersectionObserver.ts [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+type ObserverOptions = {
+  root?: React.RefObject<Element>;
+  target: React.RefObject<Element>;
+  enabled?: boolean;
+  rootMargin?: string;
+  threshold?: number | number[];
+  onIntersect(): void;
+};
+
+export function useIntersectionObserver({
```
Comment (alanuecker) [inline plugins/getty-images/src/hooks/useIntersectionObserver.ts:22]: eslint complained about the missing return type

File: plugins/getty-images/src/components/Elements/index.ts [hunk 17]
```diff
@@ -0,0 +1 @@
+export * from './Elements';
```
Comment (alanuecker) [inline plugins/getty-images/src/components/Elements/index.ts:11]: missing header in this file

File: plugins/getty-images/src/api/createAPIClient.ts [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+  refetchToken: () => void;
+  onError: OnError;
+};
+
+const refetchMaxNum = 3;
+// use refetch times constant to avoid recurcive getty requests
+// should stop refetching after refetchMaxNum
+let refetchTimes = 0;
+
+export const createAPIClient = ({
```
Comment (alanuecker) [inline plugins/getty-images/src/api/createAPIClient.ts:55]: eslint complained about the missing return type

File: plugins/getty-images/src/GettyImagesToolControlBar.tsx (hunk 12 above)
Comment (elv1n) [reply plugins/getty-images/src/GettyImagesToolControlBar.tsx]: Right, it should be skipped at all

File: plugins/getty-images/README.md [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+}
+```
+
+
+## PhotoEditor SDK Documentation
+Visit our [docs](https://docs.photoeditorsdk.com)
+
+## Examples
+[PhotoEditor SDK & Getty Images  example](https://github.com/imgly/pesdk-web-plugins/tree/main/plugins/getty-images/example)
+[PhotoEditor SDK examples](https://www.photoeditorsdk.com/)
```
Comment (alanuecker) [inline plugins/getty-images/README.md]: Is it intended that this is rendered in one line?
image

**Review by alanuecker — APPROVED** [review]

**Pull request merged**